### ISSUE TITLE
'AVAudioSession' is unavailable in macOS

### ISSUE THREAD

**Issue by tsinis** [florent37/Flutter-AssetsAudioPlayer#273]
**Flutter Version**

My version : 1.20.0-3.0.pre (both master and dev)

**Lib Version**

My version :  2.0.7+9

**Platform (Android / iOS / web) + version**

Platform : macOS (both 10.14.6 and 10.15.4)

**Describe the bug**

Compile errors: for example 'AVAudioSession' is unavailable in macOS, when building for macOS. Build fails. iOS, Android and Web versions building without errors.

**Small code to reproduce**

```dart
flutter channel dev
flutter upgrade
git clone https://github.com/tsinis/plan_et_b.git
cd plan_et_b
flutter run -d macos -v
```
[log.txt](https://github.com/florent37/Flutter-AssetsAudioPlayer/files/4874873/log.txt)

Thanks in advance!

**Comment by tsinis** [comment]
In addition, everything described above also applies to the plugin's example in this repository (https://github.com/florent37/Flutter-AssetsAudioPlayer), the same errors. In particular, I tried not to change the "macos" folder for a more relevant one (that comes with most recent "flutter create ." in Flutter's master branch) and leave the one provided in the example, but then there are even more errors at compilation. Please check by yourself:
```
flutter channel dev
flutter upgrade
git clone https://github.com/florent37/Flutter-AssetsAudioPlayer.git
cd Flutter-AssetsAudioPlayer
cd example
flutter pub get
flutter run -d macos -v
```

Thanks!

**Comment by liuchuancong** [comment]
just work on Android!  [https://github.com/florent37/Flutter-AssetsAudioPlayer/issues/259](url)

**Comment by florent37** [comment]
I'll try, maybe I forgot to just escape one "AvAudioSession" usage into in #if macos

**Comment by tsinis** [comment]
> I'll try, maybe I forgot to just escape one "AvAudioSession" usage into in #if macos

Awesome! Thanks a lot. In fact there is a several errors “not available on MacOS”. Please check the log. 

<details>
  <summary>Warnings</summary>
  

/Users/builder/programs/flutter/.pub-cache/hosted/pub.dartlang.org/assets_audio_player-2.0.7+9/macos/Classes/Music.swift:542:43: error: 'AVAudioSession' is unavailable in macOS
[        ]                                     name: AVAudioSession.interruptionNotification,
[        ]                                           ^~~~~~~~~~~~~~
[        ] AVFoundation.AVAudioSession:2:12: note: 'AVAudioSession' has been explicitly marked unavailable here
[        ] open class AVAudioSession : NSObject {
[        ]            ^
[        ] /Users/builder/programs/flutter/.pub-cache/hosted/pub.dartlang.org/assets_audio_player-2.0.7+9/macos/Classes/Music.swift:679:38: error: 'AVAudioSessionInterruptionTypeKey' is unavailable in macOS
[        ]             let typeValue = userInfo[AVAudioSessionInterruptionTypeKey] as? UInt,
[        ]                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[        ] AVFoundation.AVAudioSessionInterruptionTypeKey:2:12: note: 'AVAudioSessionInterruptionTypeKey' has been explicitly marked unavailable here
[        ] public let AVAudioSessionInterruptionTypeKey: String
[        ]            ^
[        ] /Users/builder/programs/flutter/.pub-cache/hosted/pub.dartlang.org/assets_audio_player-2.0.7+9/macos/Classes/Music.swift:680:24: error: 'AVAudioSession' is unavailable in macOS
[        ]             let type = AVAudioSession.InterruptionType(rawValue: typeValue) else {
[        ]                        ^~~~~~~~~~~~~~
[        ] AVFoundation.AVAudioSession:2:12: note: 'AVAudioSession' has been explicitly marked unavailable here
[        ] open class AVAudioSession : NSObject {
[        ]            ^
[        ] /Users/builder/programs/flutter/.pub-cache/hosted/pub.dartlang.org/assets_audio_player-2.0.7+9/macos/Classes/Music.swift:694:47: error: 'AVAudioSessionInterruptionOptionKey' is unavailable in macOS
[        ]             guard let optionsValue = userInfo[AVAudioSessionInterruptionOptionKey] as? UInt else { return }
[        ]                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[        ] AVFoundation.AVAudioSessionInterruptionOptionKey:2:12: note: 'AVAudioSessionInterruptionOptionKey' has been explicitly marked unavailable here
[        ] public let AVAudioSessionInterruptionOptionKey: String
[        ]            ^
[        ] /Users/builder/programs/flutter/.pub-cache/hosted/pub.dartlang.org/assets_audio_player-2.0.7+9/macos/Classes/Music.swift:695:27: error: 'AVAudioSession' is unavailable in macOS
[        ]             let options = AVAudioSession.InterruptionOptions(rawValue: optionsValue)
[        ]                           ^~~~~~~~~~~~~~
[        ] AVFoundation.AVAudioSession:2:12: note: 'AVAudioSession' has been explicitly marked unavailable here
[        ] open class AVAudioSession : NSObject {
[        ]            ^
[        ] /Users/builder/programs/flutter/.pub-cache/hosted/pub.dartlang.org/assets_audio_player-2.0.7+9/macos/Classes/Music.swift:698:26: error: value of type 'Player' has no member 'invokeListenerPlayPause'
[        ]                     self.invokeListenerPlayPause()
[        ]                     ~~~~ ^~~~~~~~~~~~~~~~~~~~~~~
[        ] /Users/builder/programs/flutter/.pub-cache/hosted/pub.dartlang.org/assets_audio_player-2.0.7+9/macos/Classes/Music.swift:1170:22: error: value of type 'Player' has no member 'showNotification'
[        ]                     .showNotification(show: show)
[        ]                      ^~~~~~~~~~~~~~~~


</details>

**Comment by florent37** [comment]
sorry for the delay, I just fixed it in `2.0.8` I added in this version many #if os(ios) checks
